comit/src/network/swap_digest.rs

```diff
@@ -1,19 +1,42 @@
 use crate::{asset, identity, network::protocols::announce::SwapDigest, RelativeTime, Timestamp};
-use digest::Digest;
+use digest::{Digest, ToDigestInput};
+
+// TODO should we use swap_protocols::SwapProtocol instead?
```

Comment:
Yes + deriving `Digest` is not meant for leave nodes. You should only use it on the root struct. In other words, the call to `digest` should only happen once for a particular swap.

For enums like this, implement `ToDigestInput` manually and return static bytes for each variant :)

Reply:
`swap_protocols::SwapProtocol` is rfc003 related, I do not believe it should be used.

Reply:
Ups yeah, use the new Protocol enum of course :)